### ISSUE TITLE
Bring back static pages for not signed in users.

### ISSUE THREAD

**Issue by a1batross** [mastodon/mastodon#19953]
### Steps to reproduce the problem

The current 4.0 way is to load whole dynamic frontend, which makes opening statuses on original poster's instance almost impossible on low performance computers, like mobile phones.

Personally, I've been waiting for status to render a whole minute on various instances before just giving up.

This seriously worsens User Experience on Mastodon.

### Expected behaviour

Static page should've been served.

### Actual behaviour

Nothing, except loading animation.

### Specifications

Mastodon 4.0
Android 11
Chrome 104

**Comment by retiolus** [comment]
I've used a raspberry pi with desktop these days several times to connect to various Mastodon instances and some with v4.0 and I haven't noticed any issues. Maybe the slowness you are experiencing is due to the large amount of users coming to your server and the fact that it is running at a slow rate.

**Comment by a1batross** [comment]
Mobile browsers are usually having more problems than it's desktop versions.

It could be slow connection or slow instance. New frontend makes more API
requests, which only leads to problems when Internet speed is limited or on
high packet drop rate.

Also think of a users with older browsers because their OS isn't supported
anymore. Many users nowadays disable JavaScript at all.

**Comment by trwnh** [comment]
> Many users nowadays disable JavaScript at all.

Sadly, not many users do this. But I do agree that there should be some kind of server-side rendering for no-JS users.

**Comment by spacekookie** [comment]
Yes, I'm one of those masochistic weirdos who disables JS by default. I also like to look at people's profiles on low-quality connections quite often (like on trains).

I'm not a fan of the recent public profile page redesign, but I feel like at least having a fallback to have a server-side rendered page for clients who don't support JS, would be a common courtesy to people who've been using the software in a particular way for a long time now.

**Comment by Exagone313** [comment]
My use-case here is to be able to access third-party instances without JavaScript enabled. Currently on v3, medias do not load, but it's kind of functional. With v4, you get a blank page (solid color). There is a noscript element that asks to enable JavaScript.

I get that the whole interface was rebuilt from scratch, but I think this is a terrible idea to require JavaScript on content that can be served statically to public users.

**Comment by boramalper** [comment]
I think this would also really help with SEO—not every search engine executes JavaScript (especially smaller ones). Search engines who do execute JavaScript (Google and Bing) do so within a much more constrained environment and under a tight budget, meaning higher failure rates and less content being indexed. 

Compare: `site:mastodon.social boramalper`

| Google | DuckDuckGo |
|:---:|:---:|
| ![](https://github.com/user-attachments/assets/2b94e819-71cb-4103-9749-46bc1a301c94) | ![](https://github.com/user-attachments/assets/43fc806e-4d23-4e52-85b3-caa3002701a2) |

Now if you search for `site:mastodon.social gargron` instead on DuckDuckGo, you'll get more results likely because @Gargron@mastodon.social is a lot more popular than mine. I think in a similar vein, less popular instances might be entirely missing on DuckDuckGo and other smaller search engines due to heavy requirements of JavaScript.
